packages/actor-iterator-transform-record-intermediate-results/lib/ActorIteratorTransformRecordIntermediateResults.ts

```diff
@@ -0,0 +1,52 @@
+import type {
+  IActionIteratorTransform,
+  IActionIteratorTransformBindings,
+  IActionIteratorTransformQuad,
+  IActorIteratorTransformBindingsOutput,
+  IActorIteratorTransformQuadOutput,
+  ITransformIteratorOutput }
+  from '@comunica/bus-iterator-transform';
+import { ActorIteratorTransform } from '@comunica/bus-iterator-transform';
+import { KeysStatistics } from '@comunica/context-entries';
+import type { StatisticIntermediateResults } from '@comunica/statistic-intermediate-results';
+import type { MetadataBindings, MetadataQuads } from '@comunica/types';
+import * as RDF from '@rdfjs/types';
+import { AsyncIterator } from 'asynciterator';
+
+/**
+ * A comunica Record Intermediate Results Iterator Transform Actor.
+ * This actor simply updates the intermediate result statistic when an intermediate result is produced.
+ */
+export class ActorIteratorTransformRecordIntermediateResults
+  extends ActorIteratorTransform<
+    AsyncIterator<RDF.Bindings> | AsyncIterator<RDF.Quad>,
+    MetadataBindings | MetadataQuads
+> {
+
+  public async transformIterator(action: IActionIteratorTransformQuad): 
```

Comment:
I'm not sure I understand the need for these method overrides.
AFAICS, they are not actually being used anymore?
If so, the signature from line 32 should be sufficient I guess?